### PR TITLE
Address feedback from architecture review

### PR DESCRIPTION
**Pull request by pdonahue-ventana**

### CODE REVIEW

File: introduction.tex [hunk 1]
```diff
@@ -81,12 +86,8 @@ \section{Terminology}
...
 \section{Context}
 
-\begin{steps}{This document is written to work with:}
-\item The RISC-V Instruction Set Manual, Volume I: User-Level ISA, Document
-    Version 2.2 (the ISA Spec)
-\item The RISC-V Instruction Set Manual, Volume II: Privileged Architecture,
-    Version 1.12 (the Privileged Spec)
-\end{steps}
+This document is written to work with all RISC-V ISA extensions ratified up
+through 2021.
```
[the remark this answers is not in the document]
Comment (pdonahue-ventana) [reply introduction.tex:90]: I was going to eliminate "up" but I ended up just quoting Greg directly.  I really do prefer your suggestion so let's go with that.

File: introduction.tex [hunk 2]
```diff
@@ -56,10 +56,15 @@ \section{Terminology}
     \item[JTAG]
         Refers to work done by IEEE's Joint Test Action Group, described in
         IEEE 1149.1.
+    \item[Minimal RISC-V Debug Specification]
+        The features in Sdext and the Debug Module except that the Program
+        Buffer is not required and abstract accesses to registers other than
+        GPRs, \RcsrDcsr, and \RcsrDpc are not required.  See
+        Chapter~\ref{chap:dm}.
```
[the remark this answers is not in the document]
Comment (pdonahue-ventana) [reply introduction.tex:63]: Yeah, that's simpler and it also keeps a single authoritative definition.

**Comment by pdonahue-ventana** [comment]
> Originally I had imagined we'd use Linux style version numbers: X.Y.Z X indicates really significant changes. Y is for changes that are not that significant but incompatible Z is for changes that are compatible

Policies are constantly in flux but I think that the RISC-V standard way is just X.Y.  And my understanding is that, at least for ISA specs, multiple ratified versions of an extension aren't even allowed anymore.  If you ratify Zifoo v1.0 and then you want to tweak it, you create Zibar 1.0 which is an entirely new extension which requires Zifoo (like the D floating point extension requires the F floating point extension and builds upon it).  This is not the way it was always done which is why the unpriv spec has Zicsr v2.0 and A v2.2.  But I think that's the vision going forward.

> What is our plan for clarifications etc. when we switch to a simple X.Y scheme?

I don't know if there's an official mechanism for that.  There are changes to the ratified priv and unpriv extensions all the time (5 in the last week).  There's a new date on each spec, though sometimes there are multiple new releases in the same day (so date X does not uniquely identify which version you have).

Some specs use the short git hash as part of the version ("v1.0-fd39d01" on [CMO](https://github.com/riscv/riscv-CMOs/raw/master/specifications/cmobase-v1.0.pdf)) but software would call all of those v1.0 for purposes of compiler flags or the configuration structure.  Right now the debug spec has the long git hash on the title page which is basically the same mechanism.

But then I see that CMO has a [v1.0.1](https://github.com/riscv/riscv-CMOs/commits/master) so I really don't know what's going on.  Is my description above wrong or is CMO wrong?  @gfavor?

**Comment by gfavor** [comment]
Paul is correct regarding ratified extension specs being frozen with
respect to any and all functional changes.  Only clarifications, typo
fixes, etc. are allowed.  Which the Unpriv and Priv documents (and
the extensions specs contained within them) undergo regularly.  And for
that, what Paul said is also correct.  Conversely, functional changes
require a new extension to be developed, spec'ed, and ratified.  That new
extension would typically be officially dependent on the "parent" extension
that it adds to and/or modifies.

Although in the uncommon cases like Debug where "1.0" supplants "0.13", one
could imagine a future new standalone "2.0" that supplants "1.0".  Similar
happens with Priv's Sm1p11 and Ss1p11 being supplanted by Sm1p12 and Ss1p12
(and eventually by Sm1p13 and Ss1p13).  These are all standalone
independent extensions.

So one would generally refer to the ratified XXX spec (without or without
including a version number, as appropriate).  With the understanding that
the most recent dated release of the spec is the best one to refer to since
it will contain all accumulated clarifications, typo fixes, etc.

**Comment by timsifive** [comment]
OK, so answering my own question then:
When there is a clarification to be made, we make it, update the date on the document, and don't touch the version number. That works for me.

**Review by timsifive — APPROVED** [review]

**Pull request merged**